app/models/attachable.rb

```diff
@@ -75,6 +75,14 @@ def valid_virus_state?
     true
   end
 
+  def uploaded_to_asset_manager?
+    attachments.each do |attachment|
+      next unless attachment.attachment_data
+      return false unless attachment.attachment_data.uploaded_to_asset_manager_at
```

Comment:
Might be nice to have a `AttachmentData::uploaded_to_asset_manager?` method to use here for clarity as well, then you might be able to do something like:

```ruby
attachments
  .map { |attachment| attachment.attachment_data }
  .compact
  .all? { |attachment_data| attachment_data.uploaded_to_asset_manager? }
```